src/WebApiContrib.Core/Results/HttpContextExtensions.cs

```diff
@@ -0,0 +1,549 @@
+﻿using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Net.Http.Headers;
+using System;
+using System.IO;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WebApiContrib.Core.Results
+{
+    public static class HttpContextExtensions
+    {
+        public static Task WriteActionResult<TResult>(this HttpContext context, TResult result) where TResult : IActionResult
+        {
+            var executor = context.RequestServices.GetService<IActionResultExecutor<TResult>>();
+
+            if (executor == null)
+            {
+                throw new InvalidOperationException($"No action result executor for {typeof(TResult).FullName} registered.");
+            }
+
+            var routeData = context.GetRouteData() ?? new RouteData();
```

Comment:
Should we cache these empty values?

Reply:
I thought about that, but technically they are mutable so I guess something bad might happen

Reply:
Hmm. That's a good point 👍